playlist.c

```diff
@@ -779,27 +784,45 @@ bool playlist_push(playlist_t *playlist,
 {
    size_t i;
    char real_path[PATH_MAX_LENGTH];
+   char relative_path[PATH_MAX_LENGTH];
    char real_core_path[PATH_MAX_LENGTH];
    const char *core_name = entry->core_name;
    bool entry_updated    = false;
 
    real_path[0] = '\0';
+   relative_path[0] = '\0';
    real_core_path[0] = '\0';
 
    if (!playlist || !entry)
       return false;
 
+   settings_t* settings = config_get_ptr();
+
+   /* use relative paths if enabled and entry file path is inside the content folder */
+   bool use_relative_path = (settings != NULL)
+      && settings->bools.playlist_save_relative_paths
+      && !string_is_empty(settings->paths.directory_menu_content)
+      && !string_is_empty(entry->path)
+      && (strncmp(entry->path, settings->paths.directory_menu_content, strlen(settings->paths.directory_menu_content)) == 0);
+
    if (string_is_empty(entry->core_path))
    {
       RARCH_ERR("cannot push NULL or empty core path into the playlist.\n");
       return false;
    }
 
-   /* Get 'real' path */
    if (!string_is_empty(entry->path))
    {
+      /* Get 'real' path */
       strlcpy(real_path, entry->path, sizeof(real_path));
       playlist_resolve_path(PLAYLIST_SAVE, real_path, sizeof(real_path));
+
+      if (use_relative_path)
```

Comment:
`use_relative_path` is dependent upon `!string_is_empty(entry->path)` - so you're doing `!string_is_empty(entry->path)` twice unnecessarily. Move `bool use_relative_path` inside this brace.

Reply:
(I mean, inside the `if (!string_is_empty(entry->path))` check...)